Crypto/MAC/HMAC/Conduit.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE RankNTypes, BangPatterns #-}
+-- |
+-- Module      : Crypto.MAC.HMAC.Conduit
+-- License     : BSD-style
+-- Maintainer  : Vincent Hanquez <vincent@snarc.org>
+-- Stability   : experimental
+-- Portability : unknown
+--
+-- A module containing Conduit facilities for hmac based functions.
+--
+module Crypto.MAC.HMAC.Conduit
+    ( -- * Cryptographic hash functions
+      sinkHMAC
+    ) where
+
+import Crypto.Hash
+import Crypto.MAC.HMAC
+import Data.ByteArray
+import Data.Conduit
+import Data.Void
+import qualified Data.ByteString as BS
+
+-- | A 'Sink' that calculates HMAC of a stream of 'B.ByteString'@s@ and
+-- returns digest @d@.
+sinkHMAC :: (Monad m, ByteArrayAccess key, HashAlgorithm hash) => key -> ConduitM BS.ByteString Void m (HMAC hash)
```

Comment:
It's much better to use a type variable `o` instead of `Void`, because then `sinkHMAC` can be composed in the middle of a `Conduit`. The rigid type variable here would prevent that from occurring.